podman/domain/config.py

```diff
@@ -155,19 +157,128 @@ def services(self):
-            destinations = connection.get("Connections")
-            return ServiceConnection(active, attrs=destinations[active])
+            destinations = connection.get("Connections", {})
+            if active in destinations:
+                return ServiceConnection(active, attrs=destinations[active])
 
         # if we are here, that means there was no default in the new json file
         engine = self.attrs.get("engine")
         if engine:
             active = engine.get("active_service")
-            destinations = engine.get("service_destinations")
-            return ServiceConnection(active, attrs=destinations[active])
+            destinations = engine.get("service_destinations", {})
+            if active in destinations:
+                return ServiceConnection(active, attrs=destinations[active])
 
         return None
+
+    @cached_property
+    def network_options(self) -> Dict[str, list]:
+        """
+        Retrieves network options for all configured networks.
+
+        Returns:
+            Dict[str, list]: A dictionary where keys are network names and values are lists of options.
+        """
+        network_opts = {}
+        network_config = self.attrs.get("network", {})
+        for network_name, config in network_config.items():
+            # Assuming network options are stored as 'pasta_options', 'bridge_options', etc.
+            options_key = f"{network_name}_options"
+            if options_key in network_config:
+                network_opts[network_name] = network_config[options_key]
+        return network_opts
+
+
+def create_container_with_pasta(network_name: str, port_mapping: str, image: str, container_name: str, **kwargs):
+    """
+    Creates and starts a Podman container with specified Pasta network options.
+
+    Args:
+        network_name (str): The name of the network (e.g., 'pasta').
+        port_mapping (str): The port mapping in 'host_port:container_port' format (e.g., '3128:3128').
+        image (str): The container image to use.
+        container_name (str): The name for the container.
+        **kwargs: Additional keyword arguments for container creation.
+
+    Returns:
+        podman.containers.Container: The created and started container instance.
+    """
+    try:
+        # Initialize Podman client
+        podman_client = PodmanClient(base_url="unix://run/podman/io.podman")
+        logging.debug("Podman client initialized.")
+
+        # Read Podman configuration
+        podman_config = PodmanConfig()
+        logging.debug("Podman configuration loaded.")
+
+        # Extract network options for the specified network
+        network_opts = podman_config.network_options.get(network_name, [])
+        logging.debug(f"Original network options for '{network_name}': {network_opts}")
+
+        # Append the port mapping using '-T' flag
+        network_opts += ["-T", port_mapping]
+        logging.debug(f"Updated network options for '{network_name}': {network_opts}")
```

Comment:
This looks wrong, -T maps container ports to the host and you allow only one string. Overall I don't follow this at all.